avro-converter/src/main/java/io/confluent/copycat/avro/AvroData.java

```diff
@@ -0,0 +1,1073 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.copycat.avro;
+
+import io.confluent.kafka.serializers.NonRecordContainer;
+
+import org.apache.avro.generic.GenericEnumSymbol;
+import org.apache.avro.generic.GenericFixed;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.kafka.copycat.data.Field;
+import org.apache.kafka.copycat.data.Schema;
+import org.apache.kafka.copycat.data.SchemaAndValue;
+import org.apache.kafka.copycat.data.SchemaBuilder;
+import org.apache.kafka.copycat.data.Struct;
+import org.apache.kafka.copycat.errors.DataException;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.node.ArrayNode;
+import org.codehaus.jackson.node.JsonNodeFactory;
+import org.codehaus.jackson.node.ObjectNode;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Utilities for converting between our runtime data format and Avro, and (de)serializing that data.
+ */
+public class AvroData {
+  public static final String NAMESPACE = "io.confluent.copycat.avro";
+  // Avro does not permit empty schema names, which might be the ideal default since we also are
+  // not permitted to simply omit the name. Instead, make it very clear where the default is
+  // coming from.
+  public static final String DEFAULT_SCHEMA_NAME = "CopycatDefault";
```

Comment:
Would it make sense to just force Copycat Schema to have a non-empty name? 
I can see why we may not want to force to much Avro-isms on Copycat data, but schema name isn't too convoluted.